docs/design/04-asset-records.md

```diff
@@ -0,0 +1,131 @@
+# Overview
+
+- Provide an easy way to add asset records for different physical components
+  and to assign them to nodes.
+- Should be based of `YAML` that is directly loaded into the namespace
+  (through some mechanism) and allow them to be templated against.
+- Some ability to add new asset records based of a pre-existing "templates"
+  of different asset types (from repo content)
+- The ability to edit/ delete pre-existing records
+- Asset records will need to be assigned to nodes (and reference each other
+  at some future date)
+- Some basic "validation" of the files
+
+# New Commands:
+
+There will be three basic commands:
+- `metal asset add TYPE NAME`
+- `metal asset edit NAME`
+- `metal asset delete NAME`
+
+# Adding a new asset record
+
+To make a new asset record, first a blank "template" needs to be loaded from
+repo. Note that these "templates" are not ERB. Instead they are the "YAML"
+that will provide the basis for the asset record. An no point will the
+entire "template" be rendered using ERB (however the individual tags might?).
+
+The template will then be opened using the system editor (see issues). Before
+being saved to the final location in `/var/lib/metalware/assets`. See
+`Editing` for more details on this process.
+
+In short, the validation will occur when the file is saved. This means the
+initial "templates" do not need to be valid `YAML`. Be careful of the file
+extension as it might not be `.yaml` (see issues).
```

Comment:
Is it necessary that the templates should be able to not initially be valid YAML? If we require these to be valid YAML initially, possibly with `null` entries for fields which have no other valid default, then:

- it will be easier for users initially creating/editing the template to tell they have done this correctly, as valid YAML => it's a valid template (for now);

- related to this, we can validate the template on load as well as save, which will avoid the potential of users being confused after saving a template and this failing, if the template itself contained an error they didn't spot on load;

- it will be slightly easier to initially edit a template to create a valid asset YAML file, as you can find all `null` entries in your `$EDITOR` and replace them with new values.

Reply:
Either method is easy enough to implement. It more comes down to what @ste78 and @ColonelPanicks prefers.

Reply:
I'm happy to include 'null' as default values in my templates. Will have some templates in metalware-default today.